dvc/command/ls/__init__.py

```diff
@@ -35,9 +36,10 @@ def run(self):
                 dvc_only=self.args.dvc_only,
             )
             if self.args.show_json:
-                import json
-
-                ui.write(json.dumps(entries))
+                if sys.stdout.isatty():
```

Comment:
We could hide this check under ui.write_json (if rich doesn't do that already for us) CC @skshetry

Reply:
Though that would get a little confusing with `prettify`, as it is not clear if it should enforce it or be more like "prettify, but only if it is a tty". Still, it feels like `tty` belongs to ui.

Reply:
Btw, is there any significant performace hit for prettifying? If not, maybe we should indeed consider just showing pretty json by default?

Reply:
Thanks for the comments!. It is hard to tell the performance gap between json.dumps and prettifying. I can do some evaluation on it later. Besides that, I tend to believe the prettifying is not the bottleneck since dvc list should fetch repo info from remote first which may spend more time.

Reply:
I did two simple tests on it. 
1. A python array has 100000 element, each element is a dict which contains 3 key-value.  json.dump takes about 0.05s, rich.json.from_date takes about 2.6s

2. dvc list takes about 1.8s to fetch https://github.com/iterative/example-get-started while prettifying  takes 0.03s

Reply:
I don't think performance should be a concern. We could choose a certain threshold like 1000 or so entries when we fallback to `json.dumps(indent=4)` instead of prettifying it. No one should expect us to prettify the 100000 many elements of entries.

Reply:
regretfully `dvc list`(and others) are not yet fully optimized themselves and having 100K entries in the resulting json is not unusual for our use case, so looks like prettifying is kinda expensive to do always. Plus we have other projects depending on this json output, so that migth be slowing them down too. But `isatty` check kinda solves that, because if someone will be using the json output - they will redirect it somewhere anyway, so we should be safe and efficient, while being pretty. That should make the threshold approach unnecessary.

Reply:
Thanks for the comments! I move the isatty into ui __init__.py and the patch updated. :)